sql/core/src/test/scala/org/apache/spark/sql/test/SQLTestData.scala

```diff
@@ -103,6 +103,19 @@ private[sql] trait SQLTestData { self =>
     df
   }
 
+  protected lazy val decimalDataWithNulls: DataFrame = {
```

Comment:
let's not define this here - just put it into the test case.

We wanted to get rid of SQLTestData actually, because it makes test cases much harder to read, but never had a chance to do it fully.